src/main/scala/com/evolutiongaming/crypto/DecryptConfig.scala

```diff
@@ -2,20 +2,16 @@ package com.evolutiongaming.crypto
 
 import com.typesafe.config.{Config, ConfigFactory}
 
+/**
+  * @deprecated use ConfigDecrypter which does not swallow decryption failures
+  */
+/*
+TODO: add @deprecated annotation in the next release
+@deprecated("Use ConfigDecrypter which does not swallow decryption failures", since = "3.1.0")
+ */
 object DecryptConfig {
-  private val EncryptedPasswordsPath = "encryptedPasswords"
-  private val AppSecretPath = "application.secret"
-
   def apply(password: String, config: Config = ConfigFactory.load()): String = {
-    if (
-      config.hasPath(EncryptedPasswordsPath) &&
-        config.getBoolean(EncryptedPasswordsPath)
-    ) {
-      val secret = config getString AppSecretPath
-      Crypto.decryptAES(password, secret)
-    } else {
-      password
-    }
+    ConfigDecrypter(config).decryptString(password).getOrElse(password)
```

Comment:
```suggestion
    val cd = ConfigDecrypter(config)
    if (cd.encryptedPasswordsEnabled) cd.decryptString(password).getOrElse(password)
    else password
```
This is ugly, but since you plan to deprecate this, I assume, we could survive, but this will make `ConfigDecrypter` with a better API for future use.

Reply:
The idea was to return DecryptConfig it's old semantics, roll out a replacement, do a major  release and on the next minor one deprecate the poor guy.
Doesn't your change differ from the old semantics?

Reply:
How is it different? It is more explicit, but should be the same.
if configPassword encryption disabled (else branch) -> return original value
if enabled, but fail to decrypt (if branch falling into OrElse) -> return original value
If enabled and successfully decrypted (if branch short circuit) -> return decrypted

So, for this exact method the observable outcome should be as before.

Reply:
This makes no sense at all if `decryptString` will not be changed. Perhaps we can resolve this discussion.